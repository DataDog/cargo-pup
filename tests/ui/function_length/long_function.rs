//@compile-flags: --crate-name test
//@compile-flags: --crate-type lib

// This function has more than 5 lines and should trigger a lint error
fn too_long_function() { //~ ERROR: Function exceeds maximum length of 5 lines with 9 lines
    let a = 1;
    let b = 2;
    let c = 3;
    let d = 4;
    let e = 5;
    let f = 6;
    let g = 7;
}

// This function is within the limit and should not trigger an error
fn acceptable_length() {
    let a = 1;
    let b = 2;
    let c = 3;
}

// Methods in impl blocks should also be checked
struct MyStruct;

impl MyStruct {
    // This method exceeds the length limit
    fn too_long_method(&self) { //~ ERROR: Function exceeds maximum length of 5 lines with 9 lines
        let a = 1;
        let b = 2;
        let c = 3;
        let d = 4;
        let e = 5;
        let f = 6;
        let g = 7;
    }
    
    // This method is within the limit
    fn acceptable_method(&self) {
        let a = 1;
        let b = 2;
        let c = 3;
    }
} 