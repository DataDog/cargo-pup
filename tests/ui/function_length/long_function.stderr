error: Function exceeds maximum length of 5 lines with 9 lines
  --> tests/ui/function_length/long_function.rs:5:1
   |
LL | / fn too_long_function() {
LL | |     let a = 1;
LL | |     let b = 2;
LL | |     let c = 3;
...  |
LL | |     let g = 7;
LL | | }
   | |_^
   |
   = help: 
   = note: Applied by cargo-pup rule 'long_function_test'.
   = note: `-D pup-function-length` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(pup_function_length)]`

error: Function exceeds maximum length of 5 lines with 9 lines
  --> tests/ui/function_length/long_function.rs:27:5
   |
LL | /     fn too_long_method(&self) {
LL | |         let a = 1;
LL | |         let b = 2;
LL | |         let c = 3;
...  |
LL | |         let g = 7;
LL | |     }
   | |_____^
   |
   = help: 
   = note: Applied by cargo-pup rule 'long_function_test'.

error: aborting due to 2 previous errors

